### PR TITLE
PERF cache FFTs of various quantities

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds caching of FFTs of various images in order to speed up the code.

### CODE REVIEW

**Comment by beckermr** [comment]
This one is ready for review @esheldon!

**Review by esheldon — CHANGES_REQUESTED** [review]
We should also have a speedup test for metacal, asserting it is faster by X, to justify the change.

File: ngmix/tests/test_prepsfmom.py [hunk 1]
```diff
@@ -94,6 +98,135 @@ def test_prepsfmom_raises_badjacob(cls):
...
+
+    print("%0.4f ms per fit" % (dt3/nfit*1000))
+
+    # we should never miss again for the calls above
+    assert _gauss_kernels.cache_info().misses == 2
+    assert _zero_pad_and_compute_fft_cached_impl.cache_info().misses == 4 + nfit
+
+    # if numba stuff is cached this does not work so commented out
+    # assert dt2 < dt1
+    assert dt3/nfit < dt2
```
Comment (beckermr) [inline ngmix/tests/test_prepsfmom.py:227]:
```suggestion
    assert dt3/nfit < dt2*0.6
```

**Comment by esheldon** [comment]
what about for metacal?  Is there an important and predictable speedup for this caching?

**Comment by beckermr** [comment]
The speedup comes from caching the galsim interpolated image object of the original image. It's FFT gets cached internally by galsim and this saves a bunch of time in generating the different metacal images at each color.

**Comment by beckermr** [comment]
I need to increase the digits on the jacobian repr for this one.

**Comment by beckermr** [comment]
ok @esheldon This PR is ready to go. I added a test showing the ~30% gains from caching for mdet-sized images.

**Review by esheldon — APPROVED** [review]

**Pull request merged**